api/plugins/bingo-elastic/java/src/test/java/com/epam/indigo/predicate/ReactionsTest.java

```diff
@@ -41,15 +44,20 @@ protected void check(IndigoRecordReaction target, Iterable<IndigoRecordReaction>
 
 
     protected static void loadReactions() throws Exception {
+        System.out.println("Loading reactions");
         Files.list(new File("src/test/resources/reactions/rheadb").toPath()).forEach(
                 path -> {
+                    System.out.println(path.toString());
                     IndigoRecordReaction reaction = Helpers.loadReaction(path.toString());
+                    reaction.addCustomObject("File", path.toString());
                     reactions.add(reaction);
                 }
         );
+        System.out.println("Loaded reactions count:");
+        System.out.println(reactions.size());
         Iterable<IndigoRecordReaction> reactionIterable = reactions;
         repositoryReaction.indexRecords(reactionIterable, reactions.size());
-        TimeUnit.SECONDS.sleep(5);
+        TimeUnit.SECONDS.sleep(15);
```

Comment:
Is any Sleep necessary at all? Where is asynchrony here?

Reply:
@SunFellow that's eventual consistency of Elastic plays in. One of the idea for improvements was to allow have method on repository so it would constantly check that actually documents got indexed. @khyurri @mkviatkovskii do we think it's reasonable thing to add to the API?